syntax = "proto3";

message Request {
    string inputDataLocation = 1;
    int32 numberOfMappers = 2;
    int32 numberOfReducers = 3;
    string outputDataLocation = 4;
}


message Shard {
    int64 document_id=1;
    string filename=2;
}

message Response {
    bool success = 1;
}  


message ShardList
{
    repeated Shard shards=1;
}

message RequestMapper {
    string inputDataLocation = 1;
    repeated ShardList shardlist = 2;
}

service Mapper {
    rpc map( RequestMapper) returns (Response) {}
}

service Master {
    rpc InvertedIndex (Request) returns (Response) {}
}
